From 3c7e4a2a813599a3f41cf74bdd2f5bd0eb979a9e Mon Sep 17 00:00:00 2001
From: Daniel Stone <daniels@collabora.com>
Date: Fri, 27 Jan 2017 15:11:33 +0000
Subject: [PATCH 67/94] compositor-drm: Don't restore original CRTC mode

When leaving Weston, don't attempt to restore the previous CRTC
settings. The framebuffer may well have disappeared, and in every
likelihood, whoever gets the KMS device afterwards will be repainting
anyway.

Differential Revision: https://phabricator.freedesktop.org/D1502

Signed-off-by: Daniel Stone <daniels@collabora.com>
---
 src/compositor-drm.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 9298323..aec1741 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -354,7 +354,6 @@ struct drm_output {
 	uint32_t crtc_id; /* object ID to pass to DRM functions */
 	int pipe; /* index of CRTC in resource array / bitmasks */
 	uint32_t connector_id;
-	drmModeCrtcPtr original_crtc;
 	struct drm_edid edid;
 
 	enum dpms_enum dpms;
@@ -1626,8 +1625,6 @@ drm_output_set_gamma(struct weston_output *output_base,
 	/* check */
 	if (output_base->gamma_size != size)
 		return;
-	if (!output->original_crtc)
-		return;
 
 	rc = drmModeCrtcSetGamma(backend->drm.fd,
 				 output->crtc_id,
@@ -3972,8 +3969,6 @@ create_output_for_connector(struct drm_backend *b,
 	output->pipe = i;
 	output->connector_id = connector->connector_id;
 
-	output->original_crtc = drmModeGetCrtc(b->drm.fd, output->crtc_id);
-
 	if (connector_get_current_mode(connector, b->drm.fd, &crtc_mode) < 0)
 		goto err_free;
 
@@ -4051,8 +4046,6 @@ create_output_for_connector(struct drm_backend *b,
 	output->base.assign_planes = drm_assign_planes;
 	output->base.set_dpms = drm_set_dpms;
 	output->base.switch_mode = drm_output_switch_mode;
-
-	output->base.gamma_size = output->original_crtc->gamma_size;
 	output->base.set_gamma = drm_output_set_gamma;
 
 	if (output->cursor_plane)
@@ -4096,7 +4089,6 @@ err_free:
 		free(drm_mode);
 	}
 
-	drmModeFreeCrtc(output->original_crtc);
 	free(output);
 	free(config.modeline);
 
-- 
1.9.1

